doc/sphinx/practical-tools/utilities.rst

```diff
@@ -190,6 +190,85 @@ since it results in a “packed” plugin: All auxiliary modules (as
 (``Qux_plugin``). This reduces the chances of begin unable to load two
 distinct plugins because of a clash in their auxiliary module names.
 
+.. _arbitrary_arguments:
+
+Passing arbitrary arguments to coqc
++++++++++++++++++++++++++++++++++++
+Using either ``-arg "-w all"`` or ``-arg -w -arg all`` its possible to pass the
+option ``-w all`` to ``coqc``. For options like ``-set 'Default Goal
+Selector=!'`` quotes are necessary: ``-arg "-set 'Default Goal Selector=!'"``.
+It is impossible to pass a double quote using ``-arg``, only single quotes work
+in this situation.
```

Comment:
Since the quoting rules are undocumented, one has to pass it like that, according to the doc.
This is why this PR is very welcome.

Reply:
Currently its only possible to enclose stuff in double quotes, but not in apostrophes.

Reply:
> I think this PR should also add a test in the test-suite (there is a dedicated directory for coq_makefile),

Testing for what?

Reply:
> I don't see why a user would enter the awkward `-arg -w -arg all`. Unless there is a specific case where you need this.

I also think, that this is pretty weird syntax. But I think it’s actually better behaved passing `-arg "-w all"`, since otherwise generating `COQMF_OTHERFLAGS` becomes a lot more difficult. If we’d merge the argument parsing of coqc and coq_makefile we could remove the need for `-arg`, but I think such a thing should go into a separate PR.